lang/java/reef-io/src/main/java/org/apache/reef/io/data/loading/api/DataLoadingRequestBuilder.java

```diff
@@ -188,8 +196,46 @@ public DataLoadingRequestBuilder setInputFormatClass(
     return this;
   }
 
+  /**
+   * Sets the path of the folder where the data is.
+   * Set to ANY the evaluator that will be able to load this data.
+   *
+   * @deprecated since 0.12. Should use instead
+   *             {@link DataLoadingRequestBuilder#addDataPartition(DataPartition)}
+   *             or {@link DataLoadingRequestBuilder#addDataPartitions(List)}
+   * @param inputPath
+   *          the input path
+   * @return this
+   */
+  @Deprecated
   public DataLoadingRequestBuilder setInputPath(final String inputPath) {
-    this.inputPath = inputPath;
+    this.partitions = new ArrayList<>(Arrays.asList(new DataPartition(inputPath, DataPartition.ANY)));
+    return this;
+  }
+
+  /**
+   * Adds the data partitions to the partitions list.
+   *
+   * @param dataPartitions
+   *          the data partitions to add
+   * @return this
+   */
+  public DataLoadingRequestBuilder addDataPartitions(final List<DataPartition> dataPartitions) {
```

Comment:
We should not use `List` here. We can keep the other stuff as is, but here, it should be `Iterable`, especially as code using `List`s would still work.